### PR TITLE
build(client): Update deps on build-tools to 0.39

### PR DESCRIPTION
**Pull request by jason-ha**
Updated the following:
 - client (release group)
 - client-release-group-root

Dependencies on build-tools updated:
-  `@fluid-tools/build-cli`: 0.39.0
-  `@fluidframework/build-tools`: 0.39.0
-  `@fluidframework/bundle-size-tools`: 0.39.0
-  `@fluid-tools/version-tools`: 0.39.0
- `@oclif/core` allowed to update - required by `@fluid` updates

Ran build to generate new typetest sources for changes in https://github.com/microsoft/FluidFramework/pull/19224.

- `@sealed` tag to disables "forwards" compatibility testing making extending sealed types not a breaking change. More docs are in progress about this in https://github.com/microsoft/FluidFramework/pull/14421
- Make type tests import type preprocessors instead of inlining their source. This (among other things) allows navigating to their implementations and viewing the IntelliSense documenting the different validators mentioned above.
- Add support for "MinimalType" (`@type-test-minimal`) and "FullType" (``@type-test-full``) tags to opt into stricter validation or out of validation via type preprocessors. These are mainly intended as workarounds for type-testing limitations, and an incremental step toward enabling more strict testing by default.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -359,7 +359,6 @@
 		"comments": [
 			"nodegit is replaced with an empty package here because it's currently only used by good-fences for features we do not need, and has issues building when changing node versions. See https://github.com/smikula/good-fences/issues/105 for details.",
 			"codemirror and marked overrides are because simplemde use * versions, and the fully up to date versions of its deps do not work. packageExtensions was tried to fix this, but did not work.",
-			"Fixing the @oclif/core version avoids a large number of peer dep errors since 2.6 adds a prod dep on ts-node and thus a peer dep on @types/node",
```
Comment (alexvy86) [inline package.json:362]: I imagine this is intentional; can we call out in the PR description why we're removing this now?

File: packages/common/core-interfaces/src/test/types/validateCoreInterfacesPrevious.generated.ts [hunk 2]
```diff
@@ -8,38 +8,12 @@
...
-type TypeOnly<T> = T extends number
-	? number
-	: T extends boolean | bigint | string
-		? T
-		: T extends symbol
-			? SkipUniqueSymbols<T>
-			: {
-					[P in keyof T as SkipUniqueSymbols<P>]: TypeOnly<T[P]>;
-				};
+declare type MakeUnusedImportErrorsGoAway<T> = TypeOnly<T> | MinimalType<T> | FullType<T> | typeof old | typeof current;
```
Comment (alexvy86) [inline packages/common/core-interfaces/src/test/types/validateCoreInterfacesPrevious.generated.ts:16]: Can we have a brief callout in the PR description about what improvements to type tests are brought in by build-tools 0.39?

File: packages/framework/fluid-static/src/test/types/validateFluidStaticPrevious.generated.ts [hunk 3]
```diff
@@ -142,20 +116,6 @@ declare function use_old_InterfaceDeclaration_IConnection(
...
- * typeValidation.broken:
- * "InterfaceDeclaration_IFluidContainer": {"forwardCompat": false}
- */
-declare function get_old_InterfaceDeclaration_IFluidContainer():
-    TypeOnly<old.IFluidContainer>;
-declare function use_current_InterfaceDeclaration_IFluidContainer(
-    use: TypeOnly<current.IFluidContainer>): void;
-use_current_InterfaceDeclaration_IFluidContainer(
-    get_old_InterfaceDeclaration_IFluidContainer());
-
```
Comment (alexvy86) [inline packages/framework/fluid-static/src/test/types/validateFluidStaticPrevious.generated.ts:158]: I skimmed a couple of other packages that lost type tests like this and in most cases I see they were alpha APIs, but this and IFluidContainerEvents below are public; feels like something's wrong here.

**Review by alexvy86 — COMMENTED** [review]
A couple of things below. Something feels weird about some of the typetests that went away, and removing the oclif override lets it go from 2.4 to 3.27 so just want to double check that's expected.

File: packages/common/core-interfaces/src/test/types/validateCoreInterfacesPrevious.generated.ts (hunk 2 above)
Comment (jason-ha) [reply packages/common/core-interfaces/src/test/types/validateCoreInterfacesPrevious.generated.ts:16]: @CraigMacomber / @tylerbutler, can you add such a note?

File: packages/framework/fluid-static/src/test/types/validateFluidStaticPrevious.generated.ts (hunk 3 above)
Comment (jason-ha) [reply packages/framework/fluid-static/src/test/types/validateFluidStaticPrevious.generated.ts:158]: @CraigMacomber / @tylerbutler, expected changes?

File: packages/common/core-interfaces/src/test/types/validateCoreInterfacesPrevious.generated.ts (hunk 2 above)
Comment (CraigMacomber) [reply packages/common/core-interfaces/src/test/types/validateCoreInterfacesPrevious.generated.ts:16]: Changes are from https://github.com/microsoft/FluidFramework/pull/19224, I'll add a note

Comment (CraigMacomber) [reply packages/common/core-interfaces/src/test/types/validateCoreInterfacesPrevious.generated.ts:16]: done.

File: packages/framework/fluid-static/src/test/types/validateFluidStaticPrevious.generated.ts (hunk 3 above)
Comment (CraigMacomber) [reply packages/framework/fluid-static/src/test/types/validateFluidStaticPrevious.generated.ts:158]: Yes. IFluidContainer is `@sealed` so adding items to it should not be a breaking change.

Comment (CraigMacomber) [reply packages/framework/fluid-static/src/test/types/validateFluidStaticPrevious.generated.ts:158]: Note it still includes tests to ensure functionality from it is not removed (the backwards compat tests)

**Review by tylerbutler — APPROVED** [review]

**Review by alexvy86 — APPROVED** [review]

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**